pkg/auth/user/required_attribute_verifier.go

```diff
@@ -0,0 +1,46 @@
+package user
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/sliceutils"
+)
+
+// NewRequiredAttributesVerifier returns an AttributeVerifier that will verify all
+// attributes and return an error if any of the required attributes are missing or
+// have a value that is different from expected.
+func NewRequiredAttributesVerifier(requiredAttributes []*storage.AuthProvider_RequiredAttribute) AttributeVerifier {
+	return &checkRequiredAttributesImpl{
+		required: requiredAttributes,
+	}
+}
+
+type checkRequiredAttributesImpl struct {
+	required []*storage.AuthProvider_RequiredAttribute
+}
+
+func (c *checkRequiredAttributesImpl) Verify(attributes map[string][]string) error {
+	// Attributes could be empty, handle this specifically.
+	if attributes == nil {
+		return fmt.Errorf("none of the required attributes [%s] are set", attributeKeysAsString(c.required))
+	}
+
+	for _, required := range c.required {
+		if observedValue, ok := attributes[required.GetAttributeKey()]; !ok || observedValue == nil {
+			return fmt.Errorf("missing required attribute %q", required.GetAttributeKey())
```

Comment:
Potential improvement in the future would be to collect the list of missing attributes here. Can be done in a separate PR though.

Reply:
I'll address this in a follow-up, I also wanted to refactor the test suite of `registry_httphandler` as there is some room for improvement.